From 03d3ffddd22ce3d11a6157793955b34d2c73fb27 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:39 -0400
Subject: [tools] perf/tests: Check return of perf_evlist__open sw clock event period test

Message-id: <1408462094-14194-291-git-send-email-jolsa@redhat.com>
Patchwork-id: 88295
O-Subject: [PATCH RHEL7.1 BZ1131394 290/325] perf tests: Check return of perf_evlist__open sw clock event period test
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit d0b849e9bced36cf279881294c07c43b0b9dce86
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Mon Nov 11 16:28:42 2013 -0300

description
===========
We were not checking if we successfully opened the counters, i.e. if
sys_perf_event_open worked, when it doesn't in this test, we were
continuing anyway and then segfaulting when trying to access the file
descriptor array, that at that point had been freed in perf_evlist__open
error path:

[root@ssdandy ~]# perf test -v 19
19: Test software clock events have valid period values    :
--- start ---
Segmentation fault (core dumped)
[root@ssdandy ~]#

Do the check and bail out instead.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/tests/sw-clock.c b/tools/perf/tests/sw-clock.c
index 73c5c37..ed77772 100644
--- a/tools/perf/tests/sw-clock.c
+++ b/tools/perf/tests/sw-clock.c
@@ -57,7 +57,11 @@ static int __test__sw_clock_freq(enum perf_sw_ids clock_id)
   goto out_delete_maps;
  }
 
- perf_evlist__open(evlist);
+ if (perf_evlist__open(evlist)) {
+  err = -errno;
+  pr_debug("Couldn't open evlist: %s\n", strerror(errno));
+  goto out_delete_maps;
+ }
 
  err = perf_evlist__mmap(evlist, 128, true);
  if (err < 0) {
-- 
1.7.1